File: backend/src/imports/domains/blog/entities/article_spec.py
import datetime as dt

from imports.domains.blog.entities.article import Article
from imports.domains.blog.entities.category import Category


def test_serialize_article():
    article = Article(
        slug='slug',
        title='title',
        date=dt.date(2018, 9, 9),
        categories=[
            Category(
                slug='slug 1',
                name='name 1',
            ),
            Category(
                slug='slug 2',
                name='name 2',
            )
        ],
        chicken_count=3,
        content='content',
        summary='summary',
        raw_summary='raw_summary',
    )
    assert article.serialize() == {
        'slug': 'slug',
        'title': 'title',
        'date': '2018-09-09',
        'modifiedDate': None,
        'categories': [
            category.serialize() for category in article.categories
        ],
        'chickenCount': 3,
        'content': 'content',
        'summary': 'summary',
        'rawSummary': 'raw_summary',
    }


def test_serialize_modified_article():
    article = Article(
        slug='slug',
        title='title',
        date=dt.date(2018, 9, 9),
        categories=[
            Category(
                slug='slug 1',
                name='name 1',
            ),
            Category(
                slug='slug 2',
                name='name 2',
            )
        ],
        chicken_count=3,
        content='content',
        summary='summary',
        raw_summary='raw_summary',
        modified_date=dt.date(2018, 9, 10),
    )
    assert article.serialize() == {
        'slug': 'slug',
        'title': 'title',
        'date': '2018-09-09',
        'modifiedDate': '2018-09-10',
        'categories': [
            category.serialize() for category in article.categories
        ],
        'chickenCount': 3,
        'content': 'content',
        'summary': 'summary',
        'rawSummary': 'raw_summary',
    }

File: frontend/src/imports/ui/blog/components/generic/Article/Article.spec.js
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import {MemoryRouter} from 'react-router'
import {ThemeProvider} from 'styled-components'

import Article from './Article'
import theme from '../../../theme/default'

test('It should render Article component correctly', () => {
  const article = {
    slug: 'slug',
    path: '/articles/slug',
    url: 'host://host/articles/slug',
    title: 'title',
    categories: [{
      slug: 'slug',
      name: 'name'
    }],
    date: new Date(2017, 6, 9),
    chickenCount: 1,
    content: 'content',
    rawSummary: 'raw_summary'
  }
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <MemoryRouter>
        <Article article={article}/>
      </MemoryRouter>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
